# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/02 09:40:33-08:00 rusty@rustcorp.com.au 
#   [PATCH] ipt_ECN corrupt checksum fix
#   
#   Nasty bug, caught while writing the ECN target test.  Corrupts
#   checksums of packets when target is used on them.
#   
#   Let this be a warning on the evils of casts.
#   
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# net/ipv4/netfilter/ipt_ECN.c
#   2005/01/02 02:56:47-08:00 rusty@rustcorp.com.au +1 -1
#   ipt_ECN corrupt checksum fix
# 
diff -Nru a/net/ipv4/netfilter/ipt_ECN.c b/net/ipv4/netfilter/ipt_ECN.c
--- a/net/ipv4/netfilter/ipt_ECN.c	2005-01-04 18:16:39 -08:00
+++ b/net/ipv4/netfilter/ipt_ECN.c	2005-01-04 18:16:39 -08:00
@@ -67,7 +67,7 @@
 
 	if (einfo->operation & IPT_ECN_OP_SET_CWR)
 		th->cwr = einfo->proto.tcp.cwr;
-	diffs[1] = ((u_int16_t *)&th)[6];
+	diffs[1] = ((u_int16_t *)th)[6];
 
 	/* Only mangle if it's changed. */
 	if (diffs[0] != diffs[1]) {
